### PR TITLE
Creation of Edit Student Modal- Issue #314

### PR DESCRIPTION
**Pull request by katconnors**
Link to issue:
https://github.com/sfbrigade/compass/issues/314

Styling will be in a separate issue/PR

### CODE REVIEW

**Comment by KCCPMG** [comment]
This looks like it works and my comments are pretty minor clean-up stuff. One thing I noticed (which is beyond the scope of this Issue/PR) is that you can edit a student even when there's no IEP, which is good, except that the edit form includes the fields for the IEP dates (which also don't render well when there's no dates to load). I don't consider it necessary for this PR, but it might be good to add a quick check and render those last two fields only on the condition that an IEP exists.

**Review by KCCPMG — APPROVED** [review]

File: src/pages/students/[student_id].tsx [hunk 1]
```diff
@@ -13,7 +13,32 @@ import $input from "@/styles/Input.module.css";
...
+};
+
+// uncomment and edit to apply design systems styling
+// const textfieldstyle = {
+//   border: "1px solid #20159E",
+//   width: "100%",
+// };
+
 const ViewStudentPage = () => {
+  const [open, setOpen] = React.useState(false);
```
[the remark this answers is not in the document]
Comment (katconnors) [reply src/pages/students/[student_id].tsx]: Changed!

File: src/pages/students/[student_id].tsx [hunk 2]
```diff
@@ -24,10 +49,11 @@ const ViewStudentPage = () => {
   const router = useRouter();
   const { student_id } = router.query;
 
-  const VIEW_STATES = { MAIN: 0, EDIT: 1 };
+  const VIEW_STATES = { MAIN: 0 };
```
[the remark this answers is not in the document]
Comment (katconnors) [reply src/pages/students/[student_id].tsx]: Removed this, along with two other references to view state

**Comment by katconnors** [comment]
> This looks like it works and my comments are pretty minor clean-up stuff. One thing I noticed (which is beyond the scope of this Issue/PR) is that you can edit a student even when there's no IEP, which is good, except that the edit form includes the fields for the IEP dates (which also don't render well when there's no dates to load). I don't consider it necessary for this PR, but it might be good to add a quick check and render those last two fields only on the condition that an IEP exists.

I also removed some other references to view state, so will request re-review on your end, when you have time! Thanks Connor!

File: src/pages/students/[student_id].tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (KCCPMG) [reply src/pages/students/[student_id].tsx]: I noticed the viewState and setViewState on line 46, which I think are a part of this and also outdated. I didn't even notice this before and I'll still approve

**Review by KCCPMG — APPROVED** [review]

File: src/pages/students/[student_id].tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (katconnors) [reply src/pages/students/[student_id].tsx]: Good catch! Removed this as well.

**Pull request merged**